File: modules/managed-storage/ios/ManagedStorageModule.swift
import ExpoModulesCore
import Foundation

public class ManagedStorageModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ManagedStorage")

    Function("getApplicationSupportDirectory") { () throws -> String in
      let url = try FileManager.default.url(
        for: .applicationSupportDirectory,
        in: .userDomainMask,
        appropriateFor: nil,
        create: true
      )
      return url.absoluteString
    }
  }
}

File: modules/managed-storage/android/src/main/java/expo/modules/managedstorage/ManagedStorageModule.kt
package expo.modules.managedstorage

import android.content.Context
import android.net.Uri
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ManagedStorageModule : Module() {
  private val context: Context get() = requireNotNull(appContext.reactContext)

  override fun definition() = ModuleDefinition {
    Name("ManagedStorage")

    // Internal app-specific storage: private to the app and never evicted by the system.
    Function("getApplicationSupportDirectory") {
      Uri.fromFile(context.filesDir).toString().trimEnd('/') + "/"
    }
  }
}
